tools/RAiDER/models/credentials.py

```diff
@@ -0,0 +1,90 @@
+'''
+dict that hold api information and help url for downloading weather model data
+   -  api information is stored in hidden file in home directory 
+
+        api filename      weather models          uid           key
+        cdsapirc          ERA5, ERA5T             uid           key
+        ecmwfapirc        ERAI, HRES              email         key
+        netrc             GMAO, MERRA2            username      password
+        <NAN>             HRRR [public access]    <NAN>         <NAN> 
+'''
+
+MODEL_API_DICT = {
+        'cdsapirc' :   {'api' : """\
+                                url: https://cds.climate.copernicus.eu/api/v2\
+                                \nkey: {uid}:{key}
+                                """,
+                        'url' : 'https://cds.climate.copernicus.eu/api-how-to'
+                        },
+        'ecmwfapirc' : {'api' : """{{\
+                                 \n"url"   : "https://api.ecmwf.int/v1",\
+                                 \n"key"   : "{key}",\
+                                 \n"email" : "{uid}"\
+                                 \n}}
+                                """,
+                        'url' : 'https://confluence.ecmwf.int/display/WEBAPI/Access+ECMWF+Public+Datasets#AccessECMWFPublicDatasets-key'
+                        },
+        'netrc' :       {'api' : """\
+                                \nmachine urs.earthdata.nasa.gov\
+                                \n        login {uid}\
+                                \n        password {key}\
+                                """,
+                       'url': 'https://wiki.earthdata.nasa.gov/display/EL/How+To+Access+Data+With+cURL+And+Wget'
+                       }
+                }
+
+# Function to check and write MODEL API credential for downloading weather model data
+def check_api(model: str,
+              UID: str = None,
+              KEY: str = None,
+              prompt_flag: bool = True,
+              update_flag: bool = False) -> None:
+    import os
+
+    # Weather model credential filename
+    # typically stored in home dir as hidden file
+    if model in ('ERA5', 'ERA5T'):
+        ap_filename = 'cdsapirc'
```

Comment:
typo - if you have flake8 install - should highlight this for you because this variable is never used.

Reply:
already fixed